README.md

```diff
@@ -1,2 +1,136 @@
+        let blurdroid_adapter = try!(BluetoothAdapterAndroid::get_adapter());
+        Ok(BluetoothAdapter::Android(Arc::new(blurdroid_adapter)))
+    }
+```
+On each platform you can call the same functions to reach the same GATT hierarchy elements. The following code can acces the same bluetooth device on both `android`, and `linux ` platforms:
+
+```rust
+use device::{BluetoothAdapter, BluetoothDevice};
+
+fn main() {
+    // Get the bluetooth adapter.
+    let adapter = BluetoothAdpater::init().expect("No bluetooth adapter found!");
+    // Get a device with the id 01:2A:00:4D:00:04 if it exists.
+    let device = adapter.get_device("01:2A:00:4D:00:04".to_owned() /*device address*/)
+                        .expect("No bluetooth device found!");
+}
+```
+
+#### With the *bluetooth-test* feature
+The `bluetooth-test` feature is not a default feature, to use it, append `features = ["bluetooth-test"]`, to the `device` crate dependency in the project's `Cargo.toml`. 
+
+Each enum (`BluetoothAdapter`, `BluetoothDevice`, etc.) will contain one variant of the three possible default target, and a `Mock` variant, which wraps a `Fake` structure.
+
+ android:
+```rust
+    pub enum BluetoothAdapter {
+        Android(Arc<BluetoothAdapterAndroid>),
+        Mock(Arc<FakeBluetoothAdapter>),
+    }
+```
+linux:
+```rust
+    pub enum BluetoothAdapter {
+        Bluez(Arc<BluetoothAdapterBluez>),
+        Mock(Arc<FakeBluetoothAdapter>),
+    }
+```
+
+other platforms:
+```rust
+    pub enum BluetoothAdapter {
+        Empty(Arc<BluetoothAdapterEmpty>),
+        Mock(Arc<FakeBluetoothAdapter>),
+    }
+```
+
+Beside the platform specific structures, you can create and access mock adapters, devices, services etc. These mock structures implements all the platform specific functions too. To create a mock GATT hierarchy, first you need to call the `BluetoothAdapter::init_mock()` function, insted of `BluetoothAdapter::init()`.
+
+```rust
+    use device::{BluetoothAdapter, BluetoothDevice};
+    use std::String;
+
+    // This function takes a BluetoothAdapter,
+    // and print the ids of the devices, which the adapter can find.
+    fn print_device_ids(adapter: &BluetoothAdpater) {
+        let devices = match adapter.get_devices().expect("No devices on the adapter!");
+        for device in devices {
+            println!("{:?}", device.get_id());
+        }
+    }
+
+    fn main() {
+    // This code uses a real adapter.
+        // Get the bluetooth adapter.
+        let adapter = BluetoothAdpater::init().expect("No bluetooth adapter found!");
+        // Get a device with the id 01:2A:00:4D:00:04 if it exists.
+        let device = adapter.get_device("01:2A:00:4D:00:04".to_owned() /*device address*/)
+                            .expect("No bluetooth device found!");
+
+    // This code uses a mock adapter.
+        // Creating a mock adapter.
+        let mock_adapter = BluetoothAdpater::init_mock().unwrap();
+        // Creating a mock device.
```

Comment:
It will try to create a platform specific device with the given String, and probably throws an error on Linux or Android platforms. On other platforms this will create an Empty variant.